### PR TITLE
Add basic map to FE

### LINKED ISSUE #54: Add basic map to page
Part of #34

Let's use leaflet (https://github.com/PaulLeCam/react-leaflet) and open street maps.

### PR DESCRIPTION
**Pull request by lesliexin**
**Summary**
- Closes #54 
- Added leaflet and react-leaflet to project deps 
- Added a mapbox map centered at the university of Waterloo on the Shoe Map page

image

Referenced: 
https://medium.com/javascript-in-plain-english/leaflet-map-with-react-part-i-4ef4ecbdcc1b
https://stackoverflow.com/questions/37166172/mapbox-tiles-and-leafletjs

### CODE REVIEW

File: ui/src/pages/ShoeMap.tsx [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
   return (
-    <div>
-      <TitleText>ShoeMap</TitleText>
-      <span> test text</span>
-    </div>
+    <MapContainer>
+      <TitleText>Shoe Map</TitleText>
+      <StyledMap center={currentLocation} zoom={zoom}>
+        <TileLayer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
```
Comment (aspensantos) [inline ui/src/pages/ShoeMap.tsx]: I think we discussed using mapbox instead of openstreetmap, since even though mapbox is not open source it will be free for our usage level.

File: ui/src/pages/ShoeMap.tsx [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+  height: 100%;
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
 export const ShoeMap: React.FC = () => {
+  const zoom = 12;
+  const currentLocation = { lat: 43.4723, lng: -80.5449 };
```
Comment (aspensantos) [inline ui/src/pages/ShoeMap.tsx:23]: Should these be props? If we are going to reuse this component for maps in the stories that could be helpful but if not it is probably fine like this.

File: ui/src/pages/ShoeMap.tsx (hunk 1 above)
Comment (lesliexin) [reply ui/src/pages/ShoeMap.tsx]: Ooh okay I'll change it

File: ui/src/pages/ShoeMap.tsx (hunk 2 above)
Comment (lesliexin) [reply ui/src/pages/ShoeMap.tsx:23]: Yeah for sure, I just hardcoded the values temporarily. I think I'll hold off on adding them to the props until we have a clearer idea of what we're building

File: ui/package.json [hunk 3]
```diff
@@ -25,9 +25,13 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@types/jest": "^26.0.14",
+    "@types/leaflet": "^1.5.19",
```
Comment (AbhiPrasad) [inline ui/package.json:28]: We can do this another PR, but we should move the `@types` dependencies to `devDependencies`

File: ui/src/pages/ShoeMap.tsx [hunk 4]
```diff
@@ -1,11 +1,35 @@
 import * as React from "react";
+import styled from "styled-components";
+import { Map, TileLayer } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
+
 import { TitleText } from "../styles/typography";
 
+const StyledMap = styled(Map)`
+  height: 70vh;
```
Comment (AbhiPrasad) [inline ui/src/pages/ShoeMap.tsx:9]: Took `vh`? from the examples?

File: ui/src/pages/ShoeMap.tsx [hunk 5]
```diff
@@ -1,11 +1,35 @@
 import * as React from "react";
+import styled from "styled-components";
+import { Map, TileLayer } from "react-leaflet";
+import "leaflet/dist/leaflet.css";
```
Comment (AbhiPrasad) [inline ui/src/pages/ShoeMap.tsx:4]: We should move all the css dist files to a common file and import them into `index.tsx`.

**Review by AbhiPrasad — APPROVED** [review]
Think we should be good after rebase.

File: ui/src/pages/ShoeMap.tsx (hunk 5 above)
Comment (lesliexin) [reply ui/src/pages/ShoeMap.tsx:4]: This file is imported from the leaflet package. Could you elaborate?

File: ui/src/pages/ShoeMap.tsx (hunk 4 above)
Comment (lesliexin) [reply ui/src/pages/ShoeMap.tsx:9]: No, I just added that for now

File: ui/src/pages/ShoeMap.tsx (hunk 5 above)
Comment (AbhiPrasad) [reply ui/src/pages/ShoeMap.tsx:4]: You know what, after looking at this more, I think it's fine as is, ignore my point lol.
